source/Irrlicht/COpenGLDriver.cpp

```diff
@@ -1323,14 +1323,45 @@ core::smart_refctd_ptr<IGPURenderpassIndependentPipeline> COpenGLDriver::createG
         );
 }
 
-core::smart_refctd_ptr<IGPUComputePipeline> COpenGLDriver::createGPUComputePipeline(core::smart_refctd_ptr<IGPUComputePipeline>&& _parent, core::smart_refctd_ptr<IGPUPipelineLayout>&& _layout, core::smart_refctd_ptr<IGPUSpecializedShader>&& _shader)
+core::smart_refctd_ptr<IGPUComputePipeline> COpenGLDriver::createGPUComputePipeline(IGPUPipelineCache* _pipelineCache, core::smart_refctd_ptr<IGPUComputePipeline>&& _parent, core::smart_refctd_ptr<IGPUPipelineLayout>&& _layout, core::smart_refctd_ptr<IGPUSpecializedShader>&& _shader)
 {
     if (!_layout || !_shader)
         return nullptr;
     if (_shader->getStage() != asset::ISpecializedShader::ESS_COMPUTE)
         return nullptr;
 
-    return core::make_smart_refctd_ptr<COpenGLComputePipeline>(std::move(_parent), std::move(_layout), std::move(_shader));
+    SAuxContext* ctx = getThreadContext_helper(false);
+    if (!ctx)
+        return nullptr;
+
+    GLuint GLname = 0u;
+    COpenGLSpecializedShader::SProgramBinary binary;
+    if (_pipelineCache)
+    {
+        COpenGLPipelineCache* cache = static_cast<COpenGLPipelineCache*>(_pipelineCache);
+        COpenGLPipelineLayout* layout = static_cast<COpenGLPipelineLayout*>(_layout.get());
+        COpenGLSpecializedShader* glshdr = static_cast<COpenGLSpecializedShader*>(_shader.get());
+
+        COpenGLPipelineCache::SCacheKey key{ glshdr->getSpirvHash(), glshdr->getSpecializationInfo() };
```

Comment:
layout should be in the key though? since we want to get the full driver dependent shader binary (and that depends on GLSL generated from SPIR-V which depends on the layout)

Reply:
look in code that actually finds binaries